Feature add calendar events

peterhsieh opened the pull request:
- Ability to add events to the calendar
- Added notes value for more details to add to event
- Updated with Radium (styles)
- Added Client Validation for CalendarAddEventForm
- Created two components EventList & EventItem
- New UI designed on how events are shown
- Ability to click on event and show details in Dialog window

peterhsieh committed:
Add Event Form, new API to add new events

- new API (addPropertyCalendarEvents)
- new Add Event Form modal component
- added Add Event Form component to OwnerCalendar component

TODO: fix up date/time pickers for endDateTime

peterhsieh committed:
Fixed End DateTime Pickers, adjust variables per db schema changes

- Fixed End Date/Time Picker, the same as the Start Date/Time Pickers
- Adjusted Wallfly Calendar variable to match the new db schema

peterhsieh committed:
Merge branch 'master' into feature-add-calendar-events

peterhsieh committed:
Updated with Radium

Added Radium to styles for Calendar Add Event Form

peterhsieh committed:
Changed some helper text

peterhsieh committed:
Fixed code quality, revert to calendar event having just a datetime

- Fixed camelCase for variables/properties
- Removed start/end time, stick with just a datetime
- Removed unnecessary code

- UI Changes to the calendar
- Use List (needs CSS adjustments)

peterhsieh committed:
Merge branch 'master' into feature-add-calendar-events

peterhsieh committed:
Merge branch 'master' into feature-add-calendar-events

peterhsieh committed:
Added Client Validation, Changed Calendar UI, Dialog event details

- Added Client Validation for CalendarAddEventForm
- Created two components EventList & EventItem
- Revert lots of the previous code in Wallfly Calender, new UI designed
- Ability to click on event and show details in Dialog

nirrek commented on scripts/components/CalendarAddEventForm.jsx:
```diff
@@ -0,0 +1,167 @@
+
+};
+
+// Validation schema for user profile form data.
```
Be careful when copy pasting code from other places that you don't leave obsolete comments, as this can lead to lots of confusion.

peterhsieh replied:
Ahh notes! Oops!

nirrek commented on scripts/components/CalendarAddEventForm.jsx:
```diff
@@ -0,0 +1,167 @@
+  'Event Description': Joi.string().required(),
+  Date: Joi.date().format('DD/MM/YYYY'),
+  Time: Joi.date().format('h:mm a'),
+  Notes: Joi.optional(),
```
(Same comment on the server)
You have imposed a constraint in the DB schema on the length of both notes and eventDesc. You should reflect those constraints here. Simple case, if I write a big note with >1024 characters, everything greater than 1024 characters will simply be deleted. This is a general habit to be in when programming a system. If you introduce an artificial limit (eg 1024 characters) every part of the system should reflect this limit. Otherwise you just litter the system with silent failures and bugs, which over time gets really annoying. And these things are typically easy to prevent when you introduce these constraints. "Oh I just made a hard limit on how many characters you can add, better make sure the input validation lets someone know this is the case so it doesn't silently fail on them".

nirrek commented on scripts/components/CalendarAddEventForm.jsx:
```diff
@@ -0,0 +1,167 @@
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
```
Would be best if you notify the user in the case of a server-side error while trying to process the request.

nirrek commented on scripts/components/CalendarAddEventForm.jsx:
```diff
@@ -0,0 +1,167 @@
+    ) : null;
+
+    return (
+      <div style={style.formContainer}>
```
`style.formContainer` is not defined.

nirrek commented:
It is not apparent that you can click on an event to open a more detailed description of it. Two things should be added:
- [x] When you mouse over you get the correct cursor:   `.someElement { cursor: pointer }` is what you want
- [x] You should have a subtle hover effect to highlight the event that will be opened on click.

nirrek committed:
fix syntax and unused imports

nirrek commented:
I don't think the time / date inputs have good usability.

Is the MaterialUI stuff not suitable?

If not we will need to find an alternate component we can use, as requiring exact string formats is pretty sucky usability (eg. I failed the time one the first time).

peterhsieh committed:
Adjust Validation to reflect scheme, Usability highlighting and pointer, Server Response to User

- Server response errors now put out for users
- Adjusted hint text regarding validation limits
- Adjusted validation to reflect schema
- Added highlighting on Event Items on hover

peterhsieh committed:
Filled empty space following Notes with no notes.

- Added subtle feedback to inform user that no notes were provided.
